### PR TITLE
feat: Remove all explicit intents and use safe args for search date

### LINKED ISSUE #997: Remove all activity intents
**Describe the feature you'd like**
Remove all activity intents in favor of navigation. 

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### LINKED ISSUE #998: Back stack is not clear after redirecting from order completed fragment
**Describe the bug**
The back stack is not clear when redirecting to events fragment and tickets fragment from order completed fragment.

**To Reproduce**
1. Complete an order
2. View tickets or press tick menu item 
3. Press back 

**Expected behavior**

**Would you like to work on the issue?**

- [x] Yes

### LINKED ISSUE #1047: Use type safe args for search date
**Describe the feature you'd like**
Use type safe args for search date.

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by liveHarshit**
Fixes #997  
Fixes #998
Fixes #1047

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/search/SearchViewModel.kt [hunk 1]
```diff
@@ -31,15 +31,20 @@ class SearchViewModel(
...
     private val savedNextToNextDate by lazy { preference.getString(SearchTimeViewModel.tokenKeyNextToNextDate) }
     private val savedWeekendDate by lazy { preference.getString(SearchTimeViewModel.tokenKeyWeekendDate) }
     private val savedWeekendNextDate by lazy { preference.getString(SearchTimeViewModel.tokenKeyWeekendNextDate) }
     private val savedNextMonth by lazy { preference.getString(SearchTimeViewModel.tokenKeyNextMonth) }
     private val savedNextToNextMonth by lazy { preference.getString(SearchTimeViewModel.tokenKeyNextToNextMonth) }
 
+    fun loadTimeAndLocation() {
+        savedLocation = preference.getString(tokenKey)
+        savedDate = preference.getString(SearchTimeViewModel.tokenKeyDate)
+    }
```
[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchViewModel.kt]: Store the location, no issue. Why do you need to store date?

**Comment by liveHarshit** [comment]
Updated.

**Comment by iamareebjamal** [comment]
Aren't we passing more intent extras?

**Comment by liveHarshit** [comment]
> Aren't we passing more intent extras?

yes, there are not main activity intenets anymore.

**Pull request merged**